RedHatInsights/rhsm-subscriptions
# ENT-4993 Create billable usage listener for rh-marketplace

https://issues.redhat.com/browse/ENT-4993

A new kafka consumer was added that processes BillableUsage
messages and sends the applicable usage to Red Hat Marketplace.

Both the BillableUsage and TallySummary message consumers are
attempting to send usage to RHM at the same time. This can cause
a constraint violation if subscriptions are synced when an ID is
not found. This is Ok since one of the messages will be successfully
processed. This situation is only temporary and will be addressed
when we turn off the TallySummary worker for rh-marketing.

## Testing

1. Ensure there are some Events in the DB.
2. Start the application as follows (logs get a little noisy without the grep):
```
$ LOGGING_LEVEL_ORG_CANDLEPIN_SUBSCRIPTIONS_RHMARKETPLACE=DEBUG PROM_URL="http://localhost:9090/api/v1/" SUBSCRIPTION_USE_STUB=true USER_USE_STUB=true DEV_MODE=true ./gradlew :bootRun | grep -E "Tally Summary received|Billable Usage received"
```
3. Run an hourly tally and watch the logs to make sure that there are messages getting to both listeners.
```
2022-05-25 18:44:11,287 [thread=rh-marketplace-worker-0-C-1] [DEBUG] [org.candlepin.subscriptions.rhmarketplace.RhMarketplaceWorker] - Tally Summary received by RHM for account account123!
2022-05-25 18:44:11,287 [thread=swatch-producer-rh-marketplace-0-C-1] [DEBUG] [org.candlepin.subscriptions.rhmarketplace.RhMarketplaceBillableUsageWorker] - Billable Usage received by RHM for account account123!
``` 

**NOTE:** Running without grep is a little noisy, but shows that both BillableUsage and TallySummary messages are being processed.
